### PR TITLE
use onClick property instead of handleClick for changing the type of modal rendered

### PR DESCRIPTION
**Pull request by xtina-starr**
This simplifies the `<ChangeMode>` component to just use the `onClick` property.

### CODE REVIEW

File: src/Components/Authorization/DesktopModal.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { DesktopHeader } from "./DesktopHeader"
 
 export interface DesktopModalProps extends ModalProps {
   subtitle?: string
+  onTypeChange?: (type: any) => void
```
Comment (xtina-starr) [inline src/Components/Authorization/DesktopModal.tsx]: typing this as `any` for now until we decide whether we want to use `register` or `signup`.

Comment (l2succes) [reply src/Components/Authorization/DesktopModal.tsx]: @xtina-starr let's go with `signup`. I think it's clearer then `register` for the Sign Up Form

File: src/Components/Authorization/LoginForm.tsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ export const LoginForm: FormComponentType = props => {
                 <Text color={Colors.grayDark}>Remember me</Text>
               </Checkbox>
               <ForgotPasswordLink
-                handleClick={props.handleChangeMode("reset_password")}
+                onClick={() => props.handleChangeMode("reset_password")}
```
Comment (l2succes) [inline src/Components/Authorization/LoginForm.tsx]: we should also rename `handleChangeMode` to something clearer like `presentModal`

File: src/Components/Authorization/DesktopModal.tsx (hunk 1 above)
Comment (xtina-starr) [reply src/Components/Authorization/DesktopModal.tsx]: Was leaning towards signup myself.

**Pull request merged**